frontend/src/metabase/search/components/SearchResult/SearchResult.tsx

```diff
@@ -1,126 +1,128 @@
-import { Button } from "metabase/ui";
-import { isSyncCompleted } from "metabase/lib/syncing";
-import { Icon } from "metabase/core/components/Icon";
-import Text from "metabase/components/type/Text";
+import type { LocationDescriptorObject } from "history";
+import type { MouseEvent } from "react";
+import { useCallback } from "react";
+import { push } from "react-router-redux";
 
+import { useDispatch } from "metabase/lib/redux";
 import { PLUGIN_MODERATION } from "metabase/plugins";
+import { Group, Text, Stack, Loader, Box, Divider, Title } from "metabase/ui";
+import { isSyncCompleted } from "metabase/lib/syncing";
 
 import type { WrappedResult } from "metabase/search/types";
-import Link from "metabase/core/components/Link";
-import { InfoText } from "../InfoText";
-import { ItemIcon, Context, Score } from "./components";
-import {
-  ResultButton,
-  ResultLink,
-  Title,
-  TitleWrapper,
-  Description,
-  ResultSpinner,
-  ResultLinkContent,
-  ResultInner,
-} from "./SearchResult.styled";
+import { InfoText } from "./InfoText";
+import { ItemIcon } from "./components";
+
+import { SearchResultContainer } from "./SearchResult.styled";
 
 export function SearchResult({
   result,
   compact = false,
-  hasDescription = true,
-  onClick = undefined,
+  showDescription = true,
+  onClick = null,
   isSelected = false,
 }: {
   result: WrappedResult;
   compact?: boolean;
-  hasDescription?: boolean;
-  onClick?: (result: WrappedResult) => void;
+  showDescription?: boolean;
+  onClick?: ((result: WrappedResult) => void) | null;
   isSelected?: boolean;
 }) {
-  const active = isItemActive(result);
-  const loading = isItemLoading(result);
+  const { name, model, description, moderated_status }: WrappedResult = result;
+
+  const isActive = isItemActive(result);
+  const isLoading = isItemLoading(result);
+
+  const dispatch = useDispatch();
+
+  const onChangeLocation = useCallback(
+    (nextLocation: LocationDescriptorObject | string) =>
+      dispatch(push(nextLocation)),
+    [dispatch],
+  );
 
-  // we want to remove link behavior if we have an onClick handler
-  const ResultContainer = onClick ? ResultButton : ResultLink;
+  const handleClick = (e: MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
 
-  const showXRayButton =
-    result.model === "indexed-entity" &&
-    result.id !== undefined &&
-    result.model_index_id !== null;
+    if (!isActive) {
+      return;
+    }
+
+    if (onClick) {
+      onClick(result);
+      return;
+    }
+
+    onChangeLocation(result.getUrl());
+  };
 
   return (
-    <ResultContainer
-      data-is-selected={isSelected}
+    <SearchResultContainer
```

Comment:
Just need to make this accessible. I'm working through a few ideas.

Reply:
Ok. I decided to make this a button so users will be able to navigate to this with the keyboard.

Reply:
> Ok. I decided to make this a button so users will be able to navigate to this with the keyboard.

It should be a link, because clicking it causes navigation.
If it's a link, user will be able to open the result in a new tab, copy the URL without opening the page, etc. etc.